### PR TITLE
FIX: Fix KMeans stopping condition with tolerance of exactly zero

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

Currently, passing a tolerance of exactly zero to oneDAL results in running for the specified number of maximum iterations, regardless of whether the change at some iteration is exactly zero or not. It appears that this is not how things are supposed to work on the scikit-learn side - see for example:
https://github.com/scikit-learn/scikit-learn/blob/e1fad51086a819418de3eb3a4feb08d4142d26bf/sklearn/cluster/tests/test_k_means.py#L188

---

<details>
<summary>Checklist:</summary>

**Completeness and readability**

- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

</details>

### CODE REVIEW

**Comment by david-cortes-intel** [comment]
CI errors are due to insufficient disk space, not related to this PR.

**Comment by Alexandr-Solovev** [comment]
> What's DBI?

The metric we use in benchmarks for kmeans, but if there are no changes, no necessary to recheck the metric

Thanks!
